api-gateway/src/main/java/com/petclinic/bffapigateway/domainclientlayer/InventoryServiceClient.java

```diff
@@ -61,6 +61,16 @@ public Mono<InventoryResponseDTO> updateInventory(InventoryRequestDTO model, Str
                 .accept(MediaType.APPLICATION_JSON)
                 .retrieve().bodyToMono(InventoryResponseDTO.class);
     }
+
+    public Mono<ProductResponseDTO> updateProductInInventory(final ProductRequestDTO model, final String inventoryId, final String productId){
+        return webClient
+                .put()
+                .uri(inventoryServiceUrl + "/{inventoryId}/products/{productId}", inventoryId, productId)
+                .body(Mono.just(model),ProductRequestDTO.class)
+                .accept(MediaType.APPLICATION_JSON)
+                .retrieve().bodyToMono(ProductResponseDTO.class);
```

Comment:
you should be checking the fields in the response model to make sure they are equivalent to what is expected